### PR TITLE
Link users csv with csv to docs

### PR DESCRIPTION
**Pull request by newtewt**
Now csv-to-docs can create a users csv file that works with create-users function and links to existing contact and place. Another step towards automating data seeding. 

Updated the create-users function to use roles instead of type since API deprecated type. Also, added the ability to use an existing contact or create a new one as it was doing.

### CODE REVIEW

File: src/fn/csv-to-docs.js [hunk 1]
```diff
@@ -63,7 +65,7 @@ module.exports = projectDir => {
 
     .then(() => model.references.forEach(updateRef))
     .then(() => model.exclusions.forEach(removeExcludedField))
-
+    .then(() => { if(model.user) { generateCsv(model.user,projectDir + '/users.csv'); }})
```
Comment (garethbowen) [inline src/fn/csv-to-docs.js]: Use linebreaks to make this easier to read, eg:

```js
.then(() => {
  if (model.user) {
    generateCsv(model.user,projectDir + '/users.csv');
  }
})
```

If you prefer to line all the `then`s up, you can pull out the function or include the `if` block in the `generateCsv` function.

File: src/lib/generate-users-csv.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        //passing type to the api will use that for the roles instead of roles
+        //type here is being generated based on the type of doc being created
+        delete user.type;
+    });
+    const csv = json2csvParser.parse(data);
+    fs.writeFile(csvPath, csv, function(err){
+        if (err) { throw err; }
+        info('Users csv has been saved');
+    });
+};
```
Comment (garethbowen) [inline src/lib/generate-users-csv.js]: Indentation should be 2 spaces not 4. You may need to tweak the default settings in your IDE.

File: src/fn/csv-to-docs.js [hunk 3]
```diff
@@ -104,6 +106,12 @@ module.exports = projectDir => {
       .map(r => processCsv(contactType, cols, r));
   }
 
+  function proccessUsers(csv){
```
Comment (kennsippell) [inline src/fn/csv-to-docs.js]: typo here

File: src/lib/generate-users-csv.js (hunk 2 above)
Comment (newtewt) [reply src/lib/generate-users-csv.js]: I updated a couple things to match our style-guide.

File: src/fn/csv-to-docs.js (hunk 3 above)
Comment (newtewt) [reply src/fn/csv-to-docs.js]: Thanks, I've updated.

**Pull request merged**